processor/metricstransformprocessor/metrics_transform_processor_helper.go

```diff
@@ -0,0 +1,375 @@
+				timestampToPoints[p.Timestamp.String()] = append(points, p)
+			} else {
+				timestampToPoints[p.Timestamp.String()] = []*metricspb.Point{p}
+			}
+		}
+	}
+	return timestampToPoints, startTimestamp
+}
+
+// aggregatePoints aggregates points in the groups provided in timestampToPoints by the specific caluculation indicated by aggrType and dataType
+// Returns a group of aggregated points
+func (mtp *metricsTransformProcessor) aggregatePoints(timestampToPoints map[string][]*metricspb.Point, aggrType AggregationType, dataType metricspb.MetricDescriptor_Type) []*metricspb.Point {
+	// initialize to size 0. unsure how many points will come out because distribution points are not guaranteed to be aggregatable. (mismatched bounds)
+	newPoints := make([]*metricspb.Point, 0)
+	for _, points := range timestampToPoints {
+		// use the timestamp from the first element because these all have the same timestamp as they are grouped by timestamp
+		timestamp := points[0].Timestamp
+		intPoint, doublePoint, distPoints := mtp.compute(points, aggrType, dataType)
+		if intPoint != nil {
+			newPoints = append(newPoints, &metricspb.Point{
+				Timestamp: timestamp,
+				Value:     intPoint,
+			})
+		} else if doublePoint != nil {
+			newPoints = append(newPoints, &metricspb.Point{
+				Timestamp: timestamp,
+				Value:     doublePoint,
+			})
+		} else {
+			for _, p := range distPoints {
+				newPoints = append(newPoints, &metricspb.Point{
+					Timestamp: timestamp,
+					Value:     p,
+				})
+			}
+		}
+	}
+	return newPoints
+}
+
+// compute attempts to merge points into one point based on the provided aggrType and dataType
+// Returns an int value, a double value and a slice of distribution values, and only one of these would contain values based on the dataType.
+func (mtp *metricsTransformProcessor) compute(points []*metricspb.Point, aggrType AggregationType, dataType metricspb.MetricDescriptor_Type) (*metricspb.Point_Int64Value, *metricspb.Point_DoubleValue, []*metricspb.Point_DistributionValue) {
+	intVal := int64(0)
+	doubleVal := float64(0)
+	distVals := make([]*metricspb.Point_DistributionValue, 0)
+	switch dataType {
+	case metricspb.MetricDescriptor_GAUGE_INT64, metricspb.MetricDescriptor_CUMULATIVE_INT64:
+		for _, p := range points {
+			if aggrType == Sum || aggrType == Average {
+				intVal += p.GetInt64Value()
+			} else if aggrType == Max {
+				if p.GetInt64Value() <= intVal {
+					continue
+				}
+				intVal = p.GetInt64Value()
+			}
+
+		}
+		if aggrType == Average {
+			intVal /= int64(len(points))
+		}
+		return &metricspb.Point_Int64Value{Int64Value: intVal}, nil, nil
+	case metricspb.MetricDescriptor_GAUGE_DOUBLE, metricspb.MetricDescriptor_CUMULATIVE_DOUBLE:
+		for _, p := range points {
+			if aggrType == Sum || aggrType == Average {
+				doubleVal += p.GetDoubleValue()
+			} else if aggrType == Max {
+				if p.GetDoubleValue() <= doubleVal {
+					continue
+				}
+				doubleVal = p.GetDoubleValue()
```

Comment:
`math.Max`?

Reply:
Also, I'd probably just have three totally different code paths for `sum`, `average`, & `max` since each of them are relatively straightforward independently

Reply:
Side note: I didn't notice before but it seems a little weird to support `max` but not `min`

Reply:
Min can definitely be added. I guess it just wasn't in my first version design for some reason haha. I will add that.